newIDE/app/src/ObjectsList/index.js

```diff
@@ -79,29 +110,79 @@ const objectTypeToDefaultName = {
 
 export const objectWithContextReactDndType = 'GD_OBJECT_WITH_CONTEXT';
 
-const getObjectWithContextName = (objectWithContext: ObjectWithContext) =>
-  objectWithContext.object.getName();
+const getTreeViewItemName = (item: TreeViewItem) => {
+  if (item.isRoot || item.isPlaceholder) return item.label;
+  return getObjectFolderOrObjectUnifiedName(item.objectFolderOrObject);
+};
 
-const isObjectWithContextGlobal = (objectWithContext: ObjectWithContext) =>
-  objectWithContext.global;
+const getTreeViewItemId = (item: TreeViewItem) => {
+  if (item.isRoot || item.isPlaceholder) return item.id;
+  const { objectFolderOrObject } = item;
+  if (objectFolderOrObject.isFolder()) {
+    // Use the ptr as id since two folders can have the same name.
+    // If using folder name, this would need for methods when renaming
+    // the folder to keep it open.
+    return `object-folder-${objectFolderOrObject.ptr}`;
+  }
+  const object = objectFolderOrObject.getObject();
+  // Use the ptr to avoid display bugs in the rare case a user set an object
+  // as global although another layout has an object with the same name,
+  // and ignored the warning.
+  return `${object.getName()}-${object.ptr}`;
+};
 
-const getPasteLabel = (i18n: I18nType, isGlobalObject: boolean) => {
-  let clipboardObjectName = '';
+const getTreeViewItemHtmlId = (item: TreeViewItem, index: number) =>
+  item.isRoot || item.isPlaceholder ? undefined : `object-item-${index}`;
+
+const getTreeViewItemChildren = (item: TreeViewItem) => {
+  if (item.isPlaceholder) return null;
+  if (item.isRoot) return item.children;
+  const { objectFolderOrObject, global } = item;
+  if (!objectFolderOrObject.isFolder()) return null;
+  return mapFor(0, objectFolderOrObject.getChildrenCount(), i => ({
+    objectFolderOrObject: objectFolderOrObject.getChildAt(i),
+    global,
+  }));
+};
+const getTreeViewItemData = (item: TreeViewItem) =>
+  item.isRoot || item.isPlaceholder
+    ? undefined
+    : item.objectFolderOrObject.isFolder()
```

Comment:
I believe you can move this check in the first check with a `|| item.objectFolderOrObject.isFolder()` ?